tests/configuration_tests.py

```diff
@@ -0,0 +1,119 @@
+                'test event 1': [
+                    dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                    dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                    dict(name="foobar", arg="", model="auth.User"),
+                    dict(name="", arg="invalid arg", model="auth.User"),
+                    dict(name="Not Reused Name", arg="not_reused_arg",
+                         model="auth.User"),
+                ],
+                'test event 2': [
+                    dict(name="Not Reused Name", arg="not_reused_arg",
+                         model="auth.User"),
+                    dict(foobar="abc"),
+                ],
+                'test event 3': [
+                    dict(name="User1", arg="user1", model="invalid_app.User"),
+                    dict(name="User2", arg="user2", model="auth.InvalidModel"),
+                ],
+            },
+        }
+
+        util.validate_configuration(config)
+
+        expected_messages = set([
+            "invalid event name: ''",
+            "reused data name for event 'test event 1': 'Reused Name'",
+            "reused data argument for event 'test event 1': 'reused_arg'",
+            "invalid data argument for event 'test event 1': ''",
+            "invalid data argument for event 'test event 1': 'invalid arg'",
+            "invalid data name for event 'test event 1': ''",
+            "unexpected data key for event 'test event 2': 'foobar'",
+            "missing data key for event 'test event 2': 'name'",
+            "missing data key for event 'test event 2': 'arg'",
+            "missing data key for event 'test event 2': 'model'",
+            ("bad model 'invalid_app.User' for event 'test event 3': "
+             "No installed app with label 'invalid_app'."),
+            ("bad model 'auth.InvalidModel' for event 'test event 3': "
+             "App 'auth' doesn't have a 'invalidmodel' model."),
+        ])
+        actual_messages = [message.getMessage()
+                           for message in handler.messages]
+
+        for message in actual_messages:
+            if message not in expected_messages:
+                raise self.failureException("got unexpected log message '{}'"
+                                            .format(message))
+
+            expected_messages.remove(message)
+
+        if expected_messages:
+            raise self.failureException(
+                "expected log message(s) missing: {}"
+                .format(list(expected_messages)))
+
+    @override_settings()
+    @mock.patch('django_emarsys.util.validate_configuration')
+    def test_validate_settings(self, mock_validate_configuration):
+        expected_config = {key: getattr(settings, key)
+                           for key in dir(settings)
+                           if hasattr(settings, key)}
+        util.validate_settings()
+
+        mock_validate_configuration.assert_called_with(expected_config)
```

Comment:
(Oliver answered:)
(Let's comment on the big diff of the PR instead. Can't respond without going into the commit otherwise.)

Yeah, I was unsure about the replication of the code there, but it's not entirely trivial and I want to make sure `validate_configuration()` is called correctly. It also makes sure `validate_settings()` is called at least once in tests.

Future changes to `validate_settings()` could break this test and in that case it's worth a second look at the change, I think the test is justified.

With currently 7 tests the question of test running time and maintenance is a non-issue. Let's err on the side of having a few tests too many instead of missing them.